### PR TITLE
FractionalOffset RTL

### PR DESCRIPTION
**Pull request by Hixie**
This is the first step of many down the path to RTL in the framework.

@abarth, for your critique. This is what I described the other day. Next step is to make use of it to see if it makes sense. I may apply this to EdgeInsets then try to use that one first, actually. Turns out FractionalOffset is a bit of a special snowflake in terms of how we use it.

### CODE REVIEW

File: packages/flutter/lib/src/painting/fractional_offset.dart [hunk 1]
```diff
@@ -193,5 +212,86 @@ class FractionalOffset {
...
+  ///
+  /// A value of 0.0 corresponds to the topmost edge. A value of 1.0 corresponds
+  /// to the bottommost edge. Values are not limited to that range; negative
+  /// values represent positions above the top, and values greater than 1.0
+  /// represent positions below the bottom.
+  ///
+  /// This value is passed through to [FractionalOffset.dy] unmodified by the
+  /// [resolve] method.
+  final double dy;
+
```
Comment (abarth) [inline packages/flutter/lib/src/painting/fractional_offset.dart]: What about all the operators and the static const values?

Comment (Hixie) [inline packages/flutter/lib/src/painting/fractional_offset.dart]: In the spirit of "just in time" problem solving, I'm testing the hypothesis that one only actually does maths with these once they're in the absolute coordinate space (e.g. just before painting, etc).

I suspect some static const values will make sense, but I'm similarly waiting to see what the needs are before adding them.

**Comment by abarth** [comment]
The way I would approach this problem is to do a relatively complete sketch on a branch and then use that to inform the actual patches that land on master.  For example, what you've got here is very horizontally biased.  Maybe that's the right thing, but maybe the right thing is to adhere to the maxim that "whenever there's a horizontal use case, there's also a vertical use case."

My guess is that will be easier to see once you do things like Flex, Wrap, scrolling, and grid.  For example, for Flex, I'm pretty sure it's going to be easier to support a bottom-to-top layout mode along with a right-to-level layout mode.  If that's the case, we might want to invent an enum that's similar to TextDirection but agnostic as to axis.  If we do that, we'll probably want to be able to `resolve` these directional offsets using that enum, etc.

I guess what I'm saying is that this patch looks fine as far into the future as I can see, but we might want to go further into the future to see what we want.

**Comment by Hixie** [comment]
Yup, that's my plan.

For Flex in particular, see https://github.com/flutter/flutter/issues/5089.

Current thoughts on RTL are scattered across many bugs. See https://github.com/flutter/flutter/projects/17.

**Comment by Hixie** [comment]
Going to close this for now.

**Pull request closed without merging**